### PR TITLE
DOC: Updated docs for rc_context

### PR DESCRIPTION
**Pull request by madphysicist**
Included `interactive` and `savefig` example.

### CODE REVIEW

**Comment by jenshnielsen** [comment]
Looks good. The pep8 test fails because you have a blank line with whitespace in in.

File: lib/matplotlib/__init__.py [hunk 1]
```diff
@@ -1285,7 +1285,15 @@ class rc_context(object):
             plt.plot(x, a)
 
     The 'rc' dictionary takes precedence over the settings loaded from
-    'fname'.  Passing a dictionary only is also valid.
+    'fname'.  Passing a dictionary only is also valid. For example a
+    common usage is
```
Comment (tacaswell) [inline lib/matplotlib/__init__.py]: might need trailing `::` here to make rst/sphinx happy?

**Comment by madphysicist** [comment]
Thanks. I saw the error on my phone and couldn't figure out what was causing it. Fixed now.

File: lib/matplotlib/__init__.py (hunk 1 above)
Comment (madphysicist) [inline lib/matplotlib/__init__.py]: Fixed that too

**Pull request merged**